### PR TITLE
Add tlv stream to UpdateAddHtlc message

### PR DESCRIPTION
**Pull request by btcontract**
Storing some data in `UpdateAddHtlc` will in particular help an HC peer to correctly resolve its own in-flight payments when it resyncs with incomplete local data. What's needed from Eclair is to be aware that `UpdateAddHtlc` can have some tlv data so HC plugin can store a complete message.

Currently this fails some backwards-compat tests in `wire`, not sure why, some help would be appreciated.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/UpdateAddHtlcTlv.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package fr.acinq.eclair.wire.protocol
+
+import fr.acinq.eclair.UInt64
+import fr.acinq.eclair.wire.protocol.CommonCodecs._
+import fr.acinq.eclair.wire.protocol.TlvCodecs.tlvStream
+import scodec.Codec
+import scodec.bits.ByteVector
+import scodec.codecs._
+
+object UpdateAddHtlcTlv {
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/UpdateAddHtlcTlv.scala:26]: nit: it's likely that some TLV will be shared between many `*Htlc` messages, so I would create only one file for all of them: `HtlcTlv`

File: eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/UpdateAddHtlcTlv.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  case class Data(data: ByteVector) extends Tlv
+
+  val codec: Codec[TlvStream[Data]] = tlvStream(
+    discriminated[Data].by(varint)
+      .typecase(UInt64(1L), variableSizeBytesLong(varintoverflow, bytes).as[Data])
+  )
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/UpdateAddHtlcTlv.scala:32]: We can't reserve in `eclair` tlv type `1` which isn't spec-ed.
You should create a sealed trait without any concrete implementations for now.
See what's done in `ChannelTlv` for example.

**Comment by btcontract** [comment]
Can we have some progress on this?

**Comment by t-bast** [comment]
> Can we have some progress on this?

WDYM? This PR is still draft and the build isn't passing, do you need help figuring out what's wrong?

**Comment by btcontract** [comment]
Yes, I still don't understand why specifically backwards-compatibility tests fail while otherwise it works, this is why it's still a draft.

**Comment by t-bast** [comment]
I'll try to find some time to look at this today.

**Comment by t-bast** [comment]
Closed in favor of #1891

**Pull request closed without merging**